Actions/IncrementVersionNumber/IncrementVersionNumber.psm1

```diff
@@ -0,0 +1,171 @@
+    .Parameter baseFolder
+        Base folder of the repository.
+    .Parameter project
+        Name of the project (relative to the base folder).
+    .Parameter newValue
+        New version number. If the version number starts with a +, the new version number will be added to the old version number. Else the new version number will replace the old version number.
+#>
+function Set-VersionInAppManifests($projectPath, $projectSettings, $newValue) {
+
+    # Check if the project uses repoVersion versioning strategy
+    $useRepoVersion = (($projectSettings.PSObject.Properties.Name -eq "versioningStrategy") -and (($projectSettings.versioningStrategy -band 16) -eq 16))
+    if ($useRepoVersion) {
+        $newValue = $projectSettings.repoVersion
+    }
+
+    $allAppFolders = @($projectSettings.appFolders) + @($projectSettings.testFolders) + @($projectSettings.bcptTestFolders)
+    # Set version in app.json files
+    $allAppFolders | ForEach-Object {
+        $appFolder = Join-Path $projectPath $_
+        $appJson = Join-Path $appFolder "app.json"
+
+        Set-VersionInSettingsFile -settingsFilePath $appJson -settingName 'version' -newValue $newValue
+    }
+}
+
+<#
+    .Synopsis
+        Changes the version number of dependencies in app.json files.
+    .Description
+        Changes the version number of dependencies in app.json files.
+        The version number of the dependencies is changed to the version number of the app that the dependency refers to. If the app is not found, the version number of the dependency is not changed.
+    .Parameter appFolders
+        Array of paths to the app folders. Each app folder must contain an app.json file. The apps are used to get the version number of the dependencies.
+#>
+function Set-DependenciesVersionInAppManifests {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string[]] $appFolders
+    )
+
+    # Get all apps info: app ID and app version
+    $appsInfos = @($appFolders | ForEach-Object {
+        $appJson = Join-Path $_ "app.json"
+        $app = Get-Content -Path $appJson -Raw | ConvertFrom-Json
+        return [PSCustomObject]@{
+            Id = $app.id
+            Version = $app.version
+        }
+    })
+
+    # Update dependencies in app.json files
+    $appFolders | ForEach-Object {
+        $appJsonPath = Join-Path $_ "app.json"
+
+        $appJson = Get-Content -Path $appJsonPath -Raw | ConvertFrom-Json
+
+        $dependencies = $appJson.dependencies
+
+        $dependencies | ForEach-Object {
+            $dependency = $_
+            $appInfo = $appsInfos | Where-Object { $_.Id -eq $dependency.id }
+            if ($appInfo) {
+                $dependency.version = $appInfo.Version
+            }
+        }
+
+        $appJson | ConvertTo-Json -Depth 99 | Set-Content -Path $appJsonPath
```

Comment:
Should use Set-JsonContentLF which ensures a unified json formatting
ConvertTo-Json itself formats differently on PS5 and PS7

Reply:
`Set-JsonContentLF` outputs a warning and I don't really get why:

> WARNING: pwsh (PowerShell 7) not installed, json will be formatted by PowerShell

Anyway, changed to using `Set-JsonContentLF`. The warning could be handled separately.